### PR TITLE
feat: move data plane to dedicated repository scope

### PR DESCRIPTION
**Pull request by mstasgravitee**
## :id: Reference related issue. 
https://gravitee.atlassian.net/browse/AM-3467

## :man: :woman:@mentions of the person or team responsible for reviewing proposed changes

This is a nice example: https://github.com/gravitee-io/gravitee-access-management/pull/1822

### CODE REVIEW

File: gravitee-am-repository/gravitee-am-repository-jdbc/src/test/java/io/gravitee/am/repository/management/api/PermissionTicketRepositoryPurgeTest.java
Comment (leleueri) [inline gravitee-am-repository/gravitee-am-repository-jdbc/src/test/java/io/gravitee/am/repository/management/api/PermissionTicketRepositoryPurgeTest.java:1]: **question**: why do you remove this test class ?

File: gravitee-am-repository/gravitee-am-repository-mongodb-api/src/main/java/io/gravitee/am/repository/mongodb/provider/impl/MongoConnectionProvider.java [hunk 1]
```diff
@@ -35,23 +34,28 @@
...
     private ClientWrapper<MongoClient> oauthMongoClient;
+    private ClientWrapper<MongoClient> gatewayMongoClient;
+
+    private boolean notUseOAuth2Settings;
+    private boolean notUseGatewaySettings;
 
     @Override
     public void afterPropertiesSet() throws Exception {
+        notUseOAuth2Settings = !environment.getProperty("oauth2.use-management-settings", Boolean.class, true);
+        notUseGatewaySettings = !environment.getProperty("gateway.use-management-settings", Boolean.class, true);
```
Comment (leleueri) [inline gravitee-am-repository/gravitee-am-repository-mongodb-api/src/main/java/io/gravitee/am/repository/mongodb/provider/impl/MongoConnectionProvider.java:50]:
```suggestion
        final var oauthUseMng = environment.getProperty("oauth2.use-management-settings", Boolean.class, true);
        notUseOAuth2Settings = !oauthUseMng;
        notUseGatewaySettings = !environment.getProperty("gateway.use-management-settings", Boolean.class, oauthUseMng);
```

File: gravitee-am-repository/gravitee-am-repository-jdbc/src/test/java/io/gravitee/am/repository/management/api/PermissionTicketRepositoryPurgeTest.java
Comment (mstasgravitee) [reply gravitee-am-repository/gravitee-am-repository-jdbc/src/test/java/io/gravitee/am/repository/management/api/PermissionTicketRepositoryPurgeTest.java:1]: fixed

File: gravitee-am-repository/gravitee-am-repository-mongodb-api/src/main/java/io/gravitee/am/repository/mongodb/provider/impl/MongoConnectionProvider.java (hunk 1 above)
Comment (mstasgravitee) [reply gravitee-am-repository/gravitee-am-repository-mongodb-api/src/main/java/io/gravitee/am/repository/mongodb/provider/impl/MongoConnectionProvider.java]: fixed

File: gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java [hunk 2]
```diff
@@ -47,8 +50,8 @@
 @org.springframework.context.annotation.Scope(ConfigurableBeanFactory.SCOPE_SINGLETON)
 public class R2DBCConnectionProvider implements ConnectionProvider<ConnectionFactory, R2DBCConnectionConfiguration>, InitializingBean, R2DBCSpringBeanAccessor {
 
-    @Value("${oauth2.use-management-settings:true}")
-    private boolean useManagementSettings;
+    private boolean useOAuth2Settings;
```
Comment (leleueri) [inline gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java:53]: **issue**: looks like useOAuth2Settings is never assigned

File: gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java [hunk 3]
```diff
@@ -152,22 +163,29 @@ public ConnectionProvider stop() throws Exception {
...
+        if (this.gatewayConnectionFactory != null) {
+            ((R2DBCPoolWrapper) this.gatewayConnectionFactory).shutdown();
+        }
         return this;
     }
 
     @Override
     public void afterPropertiesSet() throws Exception {
+        var useOAuth2SettingsOld = environment.getProperty("oauth2.use-management-settings", boolean.class, true);
+        useGatewaySettings = environment.getProperty("gateway.use-management-settings", boolean.class, useOAuth2SettingsOld);
```
Comment (leleueri) [inline gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java:175]: **issue**: this need to be consistent with the mongo implementation.

Let's go with the mongo implementation for now

```
        final var useOAuth2Settings = environment.getProperty("oauth2.use-management-settings", Boolean.class, true);
        final var useGatewaySettings = environment.getProperty("gateway.use-management-settings", Boolean.class, true);
        notUseOAuth2Settings = !useOAuth2Settings;
        notUseGatewaySettings = !useGatewaySettings;
```

File: gravitee-am-repository/gravitee-am-repository-mongodb-api/src/main/java/io/gravitee/am/repository/mongodb/provider/impl/MongoConnectionProvider.java [hunk 4]
```diff
@@ -35,22 +37,29 @@
...
+
+    private boolean notUseOAuth2Settings;
+    private boolean notUseGatewaySettings;
 
     @Override
     public void afterPropertiesSet() throws Exception {
+        final var useOAuth2Settings = environment.getProperty("oauth2.use-management-settings", Boolean.class, true);
+        final var useGatewaySettings = environment.getProperty("gateway.use-management-settings", Boolean.class, true);
+        notUseOAuth2Settings = !useOAuth2Settings;
+        notUseGatewaySettings = !useGatewaySettings;
```
Comment (leleueri) [inline gravitee-am-repository/gravitee-am-repository-mongodb-api/src/main/java/io/gravitee/am/repository/mongodb/provider/impl/MongoConnectionProvider.java:55]:
```suggestion
        final var useMngSettingsForOauth2 = environment.getProperty("oauth2.use-management-settings", Boolean.class, true);
        final var useMngSettingsForGateway = environment.getProperty("gateway.use-management-settings", Boolean.class, true);
        notuUseMngSettingsForOauth2 = !useMngSettingsForOauth2;
        notUseMngSettingsForGateway = !useMngSettingsForGateway;
```

File: gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java (hunk 3 above)
Comment (leleueri) [reply gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java:175]: maybe this one if you accept the proposal made on Mongo side
```
        final var useMngSettingsForOauth2 = environment.getProperty("oauth2.use-management-settings", Boolean.class, true);
        final var useMngSettingsForGateway = environment.getProperty("gateway.use-management-settings", Boolean.class, true);
        notuUseMngSettingsForOauth2 = !useMngSettingsForOauth2;
        notUseMngSettingsForGateway = !useMngSettingsForGateway;
```

**Review by leleueri — CHANGES_REQUESTED** [review]

File: gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java (hunk 2 above)
Comment (mstasgravitee) [reply gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java:53]: fixed

File: gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java (hunk 3 above)
Comment (mstasgravitee) [reply gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java:175]: fixed

File: gravitee-am-repository/gravitee-am-repository-mongodb-api/src/main/java/io/gravitee/am/repository/mongodb/provider/impl/MongoConnectionProvider.java (hunk 4 above)
Comment (mstasgravitee) [reply gravitee-am-repository/gravitee-am-repository-mongodb-api/src/main/java/io/gravitee/am/repository/mongodb/provider/impl/MongoConnectionProvider.java:55]: fixed

**Pull request merged**